binary-search.json

```diff
@@ -0,0 +1,60 @@
+{
+    "#": [
+        "Here -1 is used to indicate that the value is not included in the array.",
+        "It should be replaced with the respective expression that is idiomatic",
+        "for the language that implements the tests.",
+
+        "Following https://github.com/exercism/x-common/issues/234 the exercise",
+        "should NOT include checking whether the array is sorted as it defeats",
+        "the point of the binary search.",
+
+        "The exercise should utilize an array-like (i.e. constant-time indexed)",
+        "data structure and not a linked list. If something like an array is not",
+        "usually used in the language the problem should not be implemented.",
+        "See https://github.com/exercism/x-common/issues/244 for details."
+    ],
+    "cases": [
+        {
+            "description": "finds a value in an array with one element",
+            "array": [6],
+            "value": 6,
+            "expected": 0
+        },
+        {
+            "description": "finds a value in the middle of an array",
+            "array": [1, 3, 4, 6, 8, 9, 11],
+            "value": 6,
+            "expected": 3
+        },
+        {
+            "description": "finds a value at the beginning of an array",
+            "array": [1, 3, 4, 6, 8, 9, 11],
+            "value": 1,
+            "expected": 0
+        },
+        {
+            "description": "finds a value at the end of an array",
+            "array": [1, 3, 4, 6, 8, 9, 11],
+            "value": 11,
+            "expected": 6
+        },
+        {
+            "description": "finds a value in the middle of an array of odd length",
+            "array": [1, 3, 5, 8, 13, 34, 34, 55, 89, 144, 233, 377, 634],
```

Comment:
If there are duplicate elements the minimal case is: `[ 1, 1 ]` Is the correct answer 0 or 1?
(If 0, why is rounding up wrong and rounding down right?)

Testing implementation details can be tricky.

**IF** you do want to test implementation, then test it directly and assert that the algorithm is being implemented correctly by doing something like adding a 'depth' parameter, or asking for the array result after a certain number of iterations.

The simplest thing to do is make sure the test arrays are sorted and only contain unique values, and rely on the reviewers to call out incorrect algorithms.  This is the method used in many other exercises. (And would be my preferred method.)